tests/identity.rs

```diff
@@ -137,6 +139,58 @@ fn http2_rejects_tls_before_identity_is_certified() {
     generate_tls_reject_test! {client: client::http2_tls}
 }
 
+macro_rules! generate_outbound_tls_accept_not_cert_identity_test {
+    (server: $make_server:path, client: $make_client:path) => {
+        let _ = env_logger_init();
+        let proxy_id = "foo.ns1.serviceaccount.identity.linkerd.cluster.local";
+        let app_id = "bar.ns1.serviceaccount.identity.linkerd.cluster.local";
+
+        let proxy_identity = identity::Identity::new("foo-ns1", proxy_id.to_string());
+
+        let (_, rx) = oneshot::channel();
+        let id_svc = controller::identity().certify_async(move |_| rx).run();
+
+        let app_identity = identity::Identity::new("bar-ns1", app_id.to_string());
+        let srv = $make_server(app_identity.server_config)
+            .route("/", "hello")
+            .run();
+
+        let proxy = proxy::new()
+            .outbound(srv)
+            .identity(id_svc)
+            .run_with_test_env(proxy_identity.env);
+
+        let client = $make_client(
+            proxy.outbound,
+            app_id,
+            client::TlsConfig::new(app_identity.client_config, app_id),
+        );
+
+        assert_eventually!(
+            client
+                .request(client.request_builder("/").method("GET"))
+                .status()
+                == http::StatusCode::OK
+        );
+    };
+}
+
+#[test]
+fn http1_outbound_tls_works_before_identity_is_certified() {
+    generate_outbound_tls_accept_not_cert_identity_test! {
+    server: server::http1_tls,
+    client : client::http1_tls
+    }
+}
+
+#[test]
+fn http2_outbound_tls_works_before_identity_is_certified() {
+    generate_outbound_tls_accept_not_cert_identity_test! {
+    server: server::http2_tls,
+    client : client::http2_tls
```

Comment:
...and here?

Reply:
hmm... so rustfmt does not work for macro invocations?

Reply:
yeah, rustfmt will not format macro invocations as they're not guaranteed to contain valid Rust syntax.